src/LintDirtyModulesPlugin.js

```diff
@@ -2,6 +2,25 @@ import { isMatch } from 'micromatch';
 
 import linter from './linter';
 
+function mapFileTimestampsToOldFormat(fileTimestamps) {
+  for (const [filename, FileSystemInfoEntry] of fileTimestamps.entries()) {
+    fileTimestamps.set(
+      filename,
+      FileSystemInfoEntry ? FileSystemInfoEntry.timestamp : null
+    );
+  }
+
+  return fileTimestamps;
+}
+
+function getFileTimestamps(compilation) {
+  return (
+    compilation.fileTimestamps ||
+    // eslint-disable-next-line no-underscore-dangle
+    mapFileTimestampsToOldFormat(compilation.fileSystemInfo._fileTimestamps)
```

Comment:
`_fileTimestamps` is private. Don't do that! That's not how FileSystemInfo works. `fileSystemInfo._fileTimestamps` is a cache, while in webpack 4 `compilation.fileTimestamps` was a complete list.

Use the public functions to access timestamps of files.

Reply:
Just use `getDeprecatedFileTimestamps` from FileSystemInfo

Reply:
@sokra could we have a method in `getFileTimestamps` so as not to depend on a `getDeprecatedFileTimestamps` function that will be depreciated?

```js
getFileTimestamps() {
 return this._fileTimestamps;
}
```